### ISSUE TITLE
Override api.maptiler.com domain

### ISSUE THREAD

**Issue by tushev** [eneiluj/gpxpod#25]
Hello,

is it possible to override default api.maptiler.com domain? 

I would like to set up a proxy on my server that would perform requests to api.maptiler.com on behalf of my users, thus preserving their privacy and hiding their IPs.

**Comment by julien-nc** [comment]
It's not a straightforward thing to do. Even if you load the MapTiler style files from a different URL, it still contains resource addresses pointing to `api.maptiler.com`.

You could serve your own style.json and add it as an admin vector tile server.
To produce your style.json, just copy the content of the one loaded y GpxPod's UI (just use "style" as network request filter in the browser dev tools's network tab).
Adjust all the URLs in this style.json to point to your proxy.
Then make sure this style.json file can be loaded via the same proxy.

Is that more or less clear? :grin:

**Comment by julien-nc** [comment]
By the way, proxying raster tiles through the Nextcloud server (via a dedicated route in GpxPod) is already implemented and only used for some problematic tile servers (that have restrictive CORS policies).

There could be an admin setting to enable proxying all raster tiles through NC. What do you think?

**Comment by tushev** [comment]
> There could be an admin setting to enable proxying all raster tiles through NC. What do you think?

Yes, I suppose it would be the easiest privacy-preserving solution (provided **all** data goes through NC).
I guess, default implementation would be PHP-based, but I would prefer just to configure nginx to do proxying. 
So, it would be nice to have an option to specify your own proxy. Or at least there could be a nginx/apache rule that overrides the default implementation.

Is it also possible to rewrite all URLs specified in style.json so that all data would go through proxy? Sorry for my incompetence if that works different 😅.
